### ISSUE TITLE
license: consider MIT?

### ISSUE THREAD

**Issue by mackuba** [mapbox/mapbox-gl-native#1953]
Same issue as here: https://github.com/mapbox/mbxmapkit/issues/159

According to the current license, if I include Mapbox GL in my app, then:

> Redistributions in binary form must reproduce **the above copyright notice, this list of conditions and the following disclaimer** in the documentation and/or other materials provided with the distribution.

Which means that if I include it in an iPhone app that I put on the App Store, I need to put this long-ish text... where? In my app's App Store description, filling it with what is from my users' perspective basically garbage? Or add a separate popup screen somewhere in the app only to show this license text? Do my users really care? They (and you) only care about the fact that I do use Mapbox, but not necessarily about the complete text of the BSD license. The former is solved by having a requirement in the license terms of Mapbox the web service that I need to provide some attribution in my app, which I already do.

That's why I don't really like when people use a BSD license in open source projects, because you never know how to interpret this. Most people use a BSD license with the intention that it should be used exactly like MIT license, i.e. only keep the license with the source, even though that's not what the license actually says. But I've also seen people using BSD license specifically to create a kind of burden on people so that they have a motivation to pay them for a commercial license which allows them to not include the license in the app/docs.

So in short: please consider switching to MIT if that's what you meant :) I'm guessing that the terms of service of mapbox.com already cover what you expect app authors to do.

**Comment by kkaefer** [comment]
I fail to see how the MIT license "solves" this "problem":

> The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software.

It is sufficient to include this license and a reference to Mapbox somewhere in your terms of service/license screen. On iOS, this could e.g. be in the Settings.app.
